ui/components-react/src/components-react/newEditors/editors/EnumEditor.tsx

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as React from "react";
+import { Select } from "@itwin/itwinui-react";
+import type { EditorProps } from "../Types.js";
+import type { EnumChoice, EnumValueMetadata } from "../values/Metadata.js";
+import type { EnumValue } from "../values/Values.js";
+
+/**
+ * Enum value editor that renders `Select` component for changing value.
+ * @internal
+ */
+export function EnumEditor({
+  metadata,
+  value,
+  onChange,
+  onFinish,
+  size,
+  disabled,
+}: EditorProps<EnumValueMetadata, EnumValue>) {
+  const choices = metadata.choices;
+  const currentValue = getEnumValue(value, choices);
+
+  const handleChange = (newChoice: number | string) => {
+    const choice = choices.find((c) => c.value === newChoice);
+    const newValue = { choice: newChoice, label: choice?.label ?? "" };
+    onChange(newValue);
+    onFinish();
+  };
+
+  return (
+    <Select
```

Comment:
Does this component allow entering free-text values? If the enum is not strict, we should accept anything user enters.

Reply:
No. This works the same as an old editor. I think it would be better to add support for free-text values in a separate PR if we need it.